extensions/cl_ext_cxx_for_opencl.asciidoc

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018-2020 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+:source-highlighter: coderay
+
+= cl_ext_cxx_for_opencl
+:R: pass:q,r[^(R)^]
+Khronos{R} OpenCL Working Group
+
+== Name Strings
+
+`cl_ext_cxx_for_opencl`
+
+== Contact
+
+Please see the *Issues* list in the Khronos *OpenCL-Docs* repository: +
+https://github.com/KhronosGroup/OpenCL-Docs
+
+== Contributors
+
+Kevin Petit, Arm Ltd. +
+Sven Van Haastregt, Arm Ltd. +
+Anastasia Stulova, Arm Ltd. +
+Marco Antognini, Arm Ltd. +
+Neil Hickey, Arm Ltd. +
+
+== Notice
+
+include::../copyrights.txt[]
+
+== Status
+
+Draft
+
+== Version
+
+Built On: {docdate} +
+Version: 0.1.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification
+Version 3.0.2.
+
+This extension requires OpenCL 3.0 or OpenCL 2.x and `cl_khr_extended_versioning`.
+
+== Overview
+
+This extension adds support for building programs written using the C++ for
+OpenCL kernel language. It also enables applications to query the version of
+the language supported by the device compiler.
+
+== New build option
+
+This extension adds support for a new `CLC++` value to be passed to the
+`-cl-std` build option accepted by *clBuildProgram* and *clCompileProgram*.
+
+This option is only valid for programs created using
+*clCreateProgramFromSource*.
+
+== New API Enums
+
+Accepted value for the _param_name_ parameter to *clGetDeviceInfo*:
+
+[source,c]
+----
+CL_DEVICE_CXX_FOR_OPENCL_VERSION_EXT       0x4220
+----
+
+
+== Modifications to the OpenCL API Specification
+
+(Modify Section 4.2, *Querying Devices*) ::
++
+--
+
+(Add the following to Table 4.3, _Device Queries_) ::
++
+--
+
+[cols="1,1,4",options="header"]
+|====
+| cl_device_info
+| Return Type
+| Description
+
+| `CL_DEVICE_CXX_FOR_OPENCL_VERSION_EXT`
+| `cl_version`
+| Returns the version of the C++ for OpenCL specification implemented by the
+  device compiler.
```

Comment:
We touched on that during a call. We'd want to tag OpenCL-Docs with versions of C++ for OpenCL. It would have its own independent version number. We don't think mandating anything makes sense at this stage given the current "moving target" nature of the language. As to what the current version is, the spec currently says that it would be https://github.com/KhronosGroup/OpenCL-Docs/blob/master/cxx4opencl/diff2openclc.txt#L147 but we may want to use a smaller number.

Reply:
The current C++ for OpenCL document says "version v3.0.3-rc" at the top, so without something to override that, like the OpenCL C spec does, a reader is likely to assume that is the version number.

But for the general principal of no number while a "moving target", okay.

Reply:
Yes, that's a good point. We probably ought to change this. Thanks for flagging.

Reply:
Given the problem actually exists outside of this extension point I think this comment can be resolved by just filing an issue,

Reply:
I've created https://github.com/KhronosGroup/OpenCL-Docs/issues/420 to cover what we discussed in the last teleconference.